src/puppetlabs/jdbc_util/core.clj

```diff
@@ -190,14 +190,21 @@
       :pg_get_serial_sequence))
 
 (defn reconcile-sequence-for-column!
-  "Finds the sequence associated with the given column and sets it to the
-  current maximum value in that column, so that the next value in the sequence
-  will be the current maximum + 1 (or 0, if the table is empty). If the column
-  has no associated sequence, throws an Exception."
+  "Finds the sequence associated with the given column and sets it
+  to the greatest of:
+    * the maximum value currently in the column + 1
+    * the current value of the sequence
+    * 1 (if the column is empty)
+  If the column has no associated sequence, throws an Exception."
   [db table column]
   (if-let [sequence-name (get-sequence-name-for-column db table column)]
-    (jdbc/with-db-transaction [txn-db db]
-      (jdbc/execute! txn-db [(format "LOCK TABLE \"%s\" IN EXCLUSIVE MODE" table)])
-      (jdbc/query txn-db [(format "SELECT setval('%s', COALESCE(max(\"%s\")+1, 1), false) FROM \"%s\""
-                                  sequence-name column table)]))
+    (let [select-last-value (str "(SELECT last_value FROM " sequence-name ")")
```

Comment:
purely for readability. I find it annoying/difficult to skip back and forth between the string and the arguments at the end, especially when things get more complex.